File: category/implementation/PGS_42842_카펫.py
def solution(brown, yellow):

    total = brown + yellow
    for row in range (3, int(total**1/2) + 1):
        if total % row == 0:
            col = total // row
            if brown == 2 * (row + col - 2):
                break

    return [col,row]

File: category/implementation/PGS_42842_카펫.js
function solution(brown, yellow) {
  const total = brown + yellow;
  for (let row = 0; row < parseInt(Math.sqrt(total)) + 1; row++) {
    if (total % row == 0) {
      col = total / row;
      if (brown == 2 * (row + col - 2)) {
        return [col, row];
      }
    }
  }
}
